### PR TITLE
helm chart - Include ambassador_id when default listeners are created

### PR DESCRIPTION
**Pull request by ServerNinja**
## Description
The helm chart currently does not include the 'ambassador_id' in the default set of listeners when 'createDefaultListeners' is set as 'true' and 'AMBASSADOR_ID' exists in the environment variables. NOTE: the "module" deployed does include 'ambassador_id' as expected. This PR fixes this problem so that the two default listener objects are configured properly to work in multi ambassador installation configurations.

## Related Issues
- [Related issue](https://github.com/emissary-ingress/emissary/issues/4198)

## Testing
- Ran ' make release/push-chart' to build chart
- Installed chart on EKS cluster (adding the following to values.yaml passed in to chart):
```
env:
  AMBASSADOR_ID: "two"
```
- Verified default listeners are configured properly

## Checklist

 - [ ] I made sure to update `CHANGELOG.md`.
   
   Remember, the CHANGELOG needs to mention:
    + Any new features
    + Any changes to our included version of Envoy
    + Any non-backward-compatible changes
    + Any deprecations
 
 - [x] This is unlikely to impact how Ambassador performs at scale.
 
   Remember, things that might have an impact at scale include:
    + Any significant changes in memory use that might require adjusting the memory limits
    + Any significant changes in CPU use that might require adjusting the CPU limits
    + Anything that might change how many replicas users should use
    + Changes that impact data-plane latency/scalability
 
 - [x] My change is adequately tested.
 
   Remember when considering testing:
    + Your change needs to be specifically covered by tests.
       + Tests need to cover all the states where your change is relevant: for example, if you add a behavior that can be enabled or disabled, you'll need tests that cover the enabled case and tests that cover the disabled case. It's not sufficient just to test with the behavior enabled.
    + You also need to make sure that the _entire area being changed_ has adequate test coverage.
       + If existing tests don't actually cover the entire area being changed, add tests.
       + This applies even for aspects of the area that you're not changing – check the test coverage, and improve it if needed!
    + We should lean on the bulk of code being covered by unit tests, but...
    + ... an end-to-end test should cover the integration points
 
 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.

### CODE REVIEW

**Comment by LanceEa** [comment]
@ServerNinja - due to the current setup of our CI (hope to get fixed in the near future), I will pull your PR into a branch and verify CI is going green. 

In the meantime, can you please add an item to `charts/emissary-ingress/CHANGELOG.md`. 

Thanks,
Lance

**Comment by ServerNinja** [comment]
> @ServerNinja - due to the current setup of our CI (hope to get fixed in the near future), I will pull your PR into a branch and verify CI is going green.
> 
> In the meantime, can you please add an item to `charts/emissary-ingress/CHANGELOG.md`.
> 
> Thanks, Lance

Hi @LanceEa , I have added an item to the CHANGELOG.md. Please let me know if that works for you.

**Comment by LanceEa** [comment]
@ServerNinja  - Just one small change to the changelog and then it looks good to me. Once that is fixed I will get it merged.

Thanks again for this!

**Comment by ServerNinja** [comment]
> > > Just one small change to the changelog and then it looks good to me
> > 
> > 
> > @LanceEa , is there a specific change you need me to make to the CHANGELOG.md other than what I put in my last commit?
> 
> Yes, rather than bumping the version for the changelog can you just leave your item under the "Next Release" title? I made a comment inline with the code. The maintainers will set the version when we do another release.

Done.

**Review by LanceEa — APPROVED** [review]

**Comment by LanceEa** [comment]
@ServerNinja - I would like to get this merged for you but first we need all the commits signed-off before I can merge it. 

It looks like your second commit is missing the sign-off. Either you can squash the commits so that it is a single commit that is signed-off or feel free to just fix up the second commit so that it is signed off.

Thanks,
Lance

**Review by LanceEa — APPROVED** [review]

**Comment by LanceEa** [comment]
Looks good. I will get this merged in!

**Comment by LanceEa** [comment]
CI verified with Draft PR #4206.

**Pull request merged**
